cmake/modules/shields.cmake

```diff
@@ -48,21 +49,47 @@ set(SHIELD-NOTFOUND ${SHIELD_AS_LIST})
 
 foreach(root ${BOARD_ROOT})
   set(shield_dir ${root}/boards/shields)
-  # Match the Kconfig.shield files in the shield directories to make sure we are
-  # finding shields, e.g. x_nucleo_iks01a1/Kconfig.shield
+
+  # First, look for shield.yml files
+  file(GLOB_RECURSE shield_yml_files ${shield_dir}/*/shield.yml)
+
+  foreach(shield_yml ${shield_yml_files})
+    get_filename_component(shield_path ${shield_yml} DIRECTORY)
+    get_filename_component(shield ${shield_path} NAME)
+
+    set(yaml_ctx_name shield_data_${shield})
+    yaml_load(FILE ${shield_yml} NAME ${yaml_ctx_name})
+
+    # Check for multiple shields format first
+    yaml_get(shields_data NAME ${yaml_ctx_name} KEY shields)
+    if(NOT "${shields_data}" STREQUAL shields-NOTFOUND)
```

Comment:
should be able to do `if(shields_data)`

Reply:
Could `shield_data` have a risk of being `off`, `n`, `ignore`, or similar string ?
Cause then it could have unintended side-effects.

Reply:
I'll do however you guys think is best :) 

> Could shield_data have a risk of being off, n, ignore, or similar string ?

I don't think that would be allowed as per the schema, but then CMake doesn't really care about the schema... That being said, it wouldn't be the end of the world IMO if folks end up hitting unintended side-effects when they end up both writing invalid YAML *and* use somewhat dangerous identifiers.

Reply:
@nordicjm should be taken care of now